src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,279 @@
+
+import javax.validation.Valid;
+import javax.ws.rs.BadRequestException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.config.Constants;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.tutorialgroups.TutorialGroup;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.tutorialgroups.TutorialGroupRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.TutorialGroupService;
+import de.tum.in.www1.artemis.service.dto.StudentDTO;
+import de.tum.in.www1.artemis.service.feature.Feature;
+import de.tum.in.www1.artemis.service.feature.FeatureToggle;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+@RestController
+@RequestMapping("/api")
+public class TutorialGroupResource {
+
+    private static final String ENTITY_NAME = "tutorialGroup";
+
+    private final Logger log = LoggerFactory.getLogger(TutorialGroupResource.class);
+
+    private final TutorialGroupService tutorialGroupService;
+
+    private final TutorialGroupRepository tutorialGroupRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final UserRepository userRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    public TutorialGroupResource(AuthorizationCheckService authorizationCheckService, UserRepository userRepository, CourseRepository courseRepository,
+            TutorialGroupService tutorialGroupService, TutorialGroupRepository tutorialGroupRepository) {
+        this.tutorialGroupService = tutorialGroupService;
+        this.courseRepository = courseRepository;
+        this.userRepository = userRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.tutorialGroupRepository = tutorialGroupRepository;
+    }
+
+    /**
+     * GET /tutorial-groups/:tutorialGroupId/title : Returns the title of the tutorial-group with the given id
+     *
+     * @param tutorialGroupId the id of the tutorial group
+     * @return ResponseEntity with status 200 (OK) and with body containing the title of the tutorial group
+     */
+    @GetMapping("/tutorial-groups/{tutorialGroupId}/title")
+    @PreAuthorize("hasRole('USER')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<String> getTitle(@PathVariable Long tutorialGroupId) {
+        final var title = tutorialGroupRepository.getTutorialGroupTitle(tutorialGroupId);
+        return title == null ? ResponseEntity.notFound().build() : ResponseEntity.ok(title);
```

Comment:
`ResponseEntity.notFound()` does not offer a very descriptive error message, so we usually use an `EntityNotFoundException`, which gives a bit more detail.
But you can consider this an optional change, as it shouldn't really matter.

Reply:
Done in 5a454831567eb4226566e1e0bae347a68c357cdc